### PR TITLE
Add a separate ELF note to indicate a binary is built for CheriBSD.

### PR DESCRIPTION
**Pull request by kwitaszczyk**
**This PR depends on `__CheriBSD_version` added by #1298 and should not be merged to `dev` before #1298 is merged.**

We would like to be able to distinguish CheriBSD binaries from FreeBSD binaries. Since the 3rd-party software doesn't recognise CheriBSD as a vendor, we must leave the NT_FREEBSD_ABI_TAG note intact as it would break their build systems.

Instead, let's introduce a new note NT_CHERIBSD_ABI_TAG that indicates a binary's vendor is CheriBSD and includes a CheriBSD ABI version in its description.

I considered two changes when working on this:
* Add a second NT_FREEBSD_ABI_TAG ELF note with CheriBSD as a vendor. I'm not sure if an ELF binary can have multiple notes with the same tag. Even if so, it makes sense not to reuse the same tag since a program might use a wrong one and fail because it wouldn't recognise CheriBSD;
* Add a separate assembly file with the new note. This could organise code better but I think there's more value in having it in one file in case some changes in the upstream brand would affect our additional brand. 

This idea was proposed by @bsdjhb.

### CODE REVIEW

**Comment by kwitaszczyk** [comment]
FYI added `__CheriBSD_version` to this branch just to make sure all checks pass. I'll remove it and its dependencies before a merge.

File: lib/csu/common/notes.h
Comment (jrtc27) [inline lib/csu/common/notes.h:30]: Not sure why this changed?

File: sys/sys/elf_common.h
Comment (jrtc27) [inline sys/sys/elf_common.h:806]: The vendor is different so this can start from 1 again in its own namespace and avoid conflicts when upstream adds new ones

File: lib/csu/common/notes.h
Comment (kwitaszczyk) [reply lib/csu/common/notes.h:30]: I wanted the values to vertically align. If we prefer to reduce differences, I can revert that.

File: sys/sys/elf_common.h
Comment (kwitaszczyk) [reply sys/sys/elf_common.h:806]: I was worried that some 3rd-party software could use the last tag value instead of a pair (tag, vendor) that is recognised by it. If it's the case, it would fail with the unrecognised CheriBSD vendor name.

Comment (jrtc27) [reply sys/sys/elf_common.h:806]: Then that code is totally broken. You can't even decode the type without knowing the vendor, it can be (but normally isn't) something other than a 4-byte integer.

Comment (jrtc27) [reply sys/sys/elf_common.h:806]: (and you'd already get issues with FreeBSD notes colliding with those for NetBSD, OpenBSD, GNU, ...)

File: lib/csu/common/crtbrand.S
Comment (jrtc27) [inline lib/csu/common/crtbrand.S:50]: Shouldn't this be conditional on `__CheriBSD__` if we're going to all the trouble of defining it? Otherwise `__CheriBSD_version` won't be defined for -DWITHOUT_CHERIBSD, and even if it were you wouldn't want to include the note.

File: lib/csu/common/notes.h
Comment (kwitaszczyk) [reply lib/csu/common/notes.h]: You're right. They did but my vim setup somehow broke. After executing `:exec FreeBSD_Style()` they aligned correctly. Thanks.

File: lib/csu/common/crtbrand.S
Comment (kwitaszczyk) [reply lib/csu/common/crtbrand.S:51]: Yes, I wanted to leave it out of this review to wait for the final decision on this. Since you mentioned it, I added the ifdef.

File: sys/sys/elf_common.h
Comment (kwitaszczyk) [reply sys/sys/elf_common.h]: OK. I hope that code is not broken but we'll see when experimenting with ports.

**Review by bsdjhb — APPROVED** [review]

**Pull request merged**
